src/test/java/uk/nhs/digital/nhsconnect/nhais/model/edifact/PersonHATest.java

```diff
@@ -0,0 +1,35 @@
+package uk.nhs.digital.nhsconnect.nhais.model.edifact;
+
+import org.junit.jupiter.api.Test;
+import uk.nhs.digital.nhsconnect.nhais.model.edifact.message.EdifactValidationException;
+
+import static org.junit.jupiter.api.Assertions.assertEquals;
+import static org.junit.jupiter.api.Assertions.assertThrows;
+
+public class PersonHATest {
+
+    @Test
+    public void When_MappingToEdifact_Then_ReturnCorrectString() {
+        var expectedValue = "NAD+FHS+XX1:954'";
+
+        var personHA = PersonHA.builder()
+            .organization("XX1")
+            .build();
+
+        assertEquals(expectedValue, personHA.toEdifact());
+    }
+
+    @Test
+    public void When_MappingToEdifactWithEmptyHA_Then_EdifactValidationExceptionIsThrown() {
+        var personHA = PersonHA.builder()
+            .organization("")
+            .build();
+
+        assertThrows(EdifactValidationException.class, personHA::toEdifact);
+    }
+
+    @Test
+    public void When_BuildingWithoutOrganization_Then_NullPointerExceptionIsThrown() {
+        assertThrows(NullPointerException.class, () -> PersonHA.builder().build());
```

Comment:
Use different/custom exception instead of NPE (update also other segment test classes)

Reply:
It's thrown by Lombok when an object is built without mandatory field like:
private @NonNull String organization;

Reply:
ok